### DISCUSSION TITLE
"TypeError: Cannot read property '_json' of undefined" with JEST + jest-openapi + PactumJS

### DISCUSSION THREAD

**Discussion by YauheniN** [pactumjs/pactum#143] [Q&A]
Hello! Now I'm using supertest + jest-openapi for asserting that HTTP responses satisfy an OpenAPI spec. But I mainly plan to use PactumJS on a project for others assertions and have idea also use for this PactumJS. I'm trying to create assertion OpenApi like this:

```js 
const res = await pactum.spec().get('/account/stats').expectStatus(200);
await expect(await res).toSatisfyApiSpec();
```

And getting error: "TypeError: Cannot read property '_json' of undefined" .  How send request or prepare response with PactumJS for assertion OpenAPI spec?

**Comment by ASaiAnudeep** [comment]
I have gone through the docs of [jest-openapi](https://www.npmjs.com/package/jest-openapi). Seems the response object accepted by `jest-openapi` is different from what `pactum` returns. 

This is an interesting use case and I would like to see `pactum` supporting these kind of assertions by default. I'm marking this as a feature request.

As a work around, this might do the trick.

```js
const res = await pactum.spec().get('/account/stats').expectStatus(200);
res['_json '] = res.json;
await expect(await res).toSatisfyApiSpec();
```

**Reply by YauheniN** [reply]
Thanks for marking this as a feature request. It will be excellent). But your decision doesn't work in my case, I tried it earlier. Will be waiting support these kind of assertions by default.

**Reply by txm** [reply]
Did you find the answer?  I'm not sure it is related to PactumJS as I'm not using and have the same problem on calling .toSatisfyApiSpec().
